cmd/thanos/receive.go

```diff
@@ -258,6 +259,15 @@ func runReceive(
 	// hashringChangedChan signals when TSDB needs to be flushed and updated due to hashring config change.
 	hashringChangedChan := make(chan struct{}, 1)
 
+	level.Debug(logger).Log("msg", "setting up hashring")
+	{
+		if hashringConfigs, err := setupHashring(g, logger, reg, conf, hashringChangedChan, webHandler, statusProber, enableIngestion); err != nil {
```

Comment:
Like we discussed, maybe this isn't a great approach, as it creates a dependency between hashring and tsdb components on startup